### PR TITLE
Unit 13 Outlook : Chapter about Hiera

### PR DESCRIPTION
**Pull request by dariowsz**
This chapter discusses the main takeaways of the Hiera paper published by Meta AI. The following sections are covered:

- **What is Hiera?**: Introduction to the Hiera architecture.
- **From CNNs to ViTs**: Short recap on the evolution of the computer vision architectures which motivates the creation of Hiera.
- **Hiera's approach: Pretraining task is all you need**: How Hiera manages to achive great performance and speed without overengineering the model's architecture
- **MAE**: Short introduction to the pretraining task used by Hiera.
- **Why it matters?**: What does the creation of Hiera implies in the broader context of machine learning and AI.

Feel free to leave suggestions!

### CODE REVIEW

File: chapters/en/Unit 13 - Outlook/hiera.mdx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+## Overview
+
+### What is Hiera?
+
+Hiera is a hierarchical [vision transformer](https://arxiv.org/abs/2010.11929) that achieves high accuracy without the need for specialized components found in other vision models. The authors propose pretraining Hiera with a strong visual pretext task to remove unnecessary complexity and create a faster and more accurate model.
```
Comment (mmhamdy) [inline chapters/en/Unit 13 - Outlook/hiera.mdx:5]:
```suggestion
Hiera is a hierarchical [vision transformer](https://arxiv.org/abs/2010.11929) that achieves high accuracy without the need for specialized components found in other hierarchical vision models. The authors propose pretraining Hiera with a strong visual pretext task to remove unnecessary complexity and create a faster and more accurate model.
```

File: chapters/en/Unit 13 - Outlook/hiera.mdx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+### Hiera's approach: Pretraining task is all you need
+
+Authors of the Hiera paper argue that a ViT model can learn spatial reasoning and perform well on computer vision tasks by using a strong visual pretext task called MAE and thus, they can remove unnecessary components and complexity from state-of-the-art multi-stage vision transformers to achieve greater accuracy and speed.
+
+### MAE
+MAE, which means “Masked AutoEncoder”, is an unsupervised training paradigm. As any other Autoencoder, it consists of encoding high dimensional data (images) into a lower dimension representation (embeddings), in such a way that this data can be decoded into the original high dimensional data again. However, the visual MAE technique consists of dropping certain amount of patches (around 75%), encoding the rest of the patches, and then trying to predict the missing ones. This idea is based on the BERT model, and has been used a lot in recent years as a pre-training task for image encoders.
```
Comment (mmhamdy) [inline chapters/en/Unit 13 - Outlook/hiera.mdx:24]:
```suggestion
MAE, short for “Masked AutoEncoder”, in this context, is an unsupervised training paradigm. As with any other method using Autoencoders, it consists of encoding high dimensional data (images) into a lower dimension representation (embeddings), in such a way that this data can be decoded into the original high dimensional data again. However, the visual MAE technique consists of dropping a certain amount of patches (around 75%), encoding the rest of the patches, and then trying to predict the missing ones. This idea is based on the BERT model and has been used a lot in recent years as a pre-training task for image encoders.
```
I think it'd be nice to talk about masked pretraining in general

**Review by mmhamdy — COMMENTED** [review]
Thanks, @dariowsz. Looks amazing! But I think this chapter should include at least a passage about hierarchical architectures and the components they add.

**Comment by gonzachiar** [comment]
> Thanks, @dariowsz. Looks amazing! But I think this chapter should include at least a passage about hierarchical architectures and the components they add.

Hi @mmhamdy, I'm working on this PR with @dariowsz.

We were thinking of doing a summary of every architecture change from MViT-v2 to Hiera, basically every component on this list:

Screenshot 2024-01-02 at 16 05 55

(Comment: we were thinking of merging b and c into one point since both are part of the same change)

On the other side, we agree with your comment about Masked training. However, we would like to do both: an introduction to masked training and a focus on MAE. We are not sure how deep to go into this, since there are other chapters that may talk about masked training (BERT, ELMO, or GPT), and we don't know if it is okay if we overlap with them.

**Comment by mmhamdy** [comment]
That sounds super nice, @gonzachiar! Concerning masked training, feel free to add as much as you like (constrained by the context of computer vision, of course).

**Comment by gonzachiar** [comment]
Awesome! We were thinking on updating this PR on January 10, is that okay?

**Comment by mmhamdy** [comment]
Sure, take your time.

**Review by merveenoyan — COMMENTED** [review]
Mostly nits, thanks a lot!!! 🤗

**Review by merveenoyan — APPROVED** [review]
LGTM

**Comment by merveenoyan** [comment]
@dariowsz would you like to add a small implementation of the Hiera architecture?

**Review by johko — APPROVED** [review]
Nice write-up, definitely learned a bit here :+1:

**Comment by merveenoyan** [comment]
@dariowsz you can raise another PR for the implementation, just merged this.

**Comment by dariowsz** [comment]
Great! Thank you all for reviewing our work! Will try to code an implementation and maybe add some extra details in another PR

**Pull request merged**